### PR TITLE
Switch build backend to hatchling

### PR DESCRIPTION
**Pull request by cbrnr**
- [x] Switch build backend to hatchling
- [x] Switch to src layout
- [x] Use `python -m build` as a build frontend in workflows
- [x] Use trusted publishing on PyPI

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -54,6 +57,9 @@ documentation = "https://mnelab.readthedocs.io/en/latest/"
 repository = "https://github.com/cbrnr/mnelab"
 changelog = "https://github.com/cbrnr/mnelab/blob/main/CHANGELOG.md"
 
+[tool.hatch.version]
+path = "src/mnelab/__init__.py"
```
Comment (hoechenberger) [inline pyproject.toml:61]: why is this needed?

Comment (cbrnr) [reply pyproject.toml:61]: Because that's where the version is defined. This will become obsolete when I switch to `hatch-vcs` though.

**Review by cbrnr — COMMENTED** [review]
Merging, let's tackle `hatch-vcs` in another PR.

**Comment by hoechenberger** [comment]
@cbrnr I believe you forgot to remove `MANIFEST.in`
